django/core/validators.py

```diff
@@ -94,7 +94,7 @@ class URLValidator(RegexValidator):
 
     regex = _lazy_re_compile(
         r'^(?:[a-z0-9\.\-\+]*)://'  # scheme is validated separately
-        r'(?:\S+(?::\S*)?@)?'  # user:pass authentication
+        r'(?:[^\s:@/]+(?::[^\s:@/]*)?@)?'  # user:pass authentication
```

Comment:
Is it worth adding an invalid URL for each of the cases here (beyond `/` which is already added)?

Reply:
I don't feel particularly strongly either way.

Reply:
I've added some test cases that should cover all possible regressions in this regex change.